include/swift/Basic/LangOptions.h

```diff
@@ -346,6 +346,8 @@ namespace swift {
                  Target.isPS4() || Target.isOSHaiku() ||
                  Target.getTriple().empty()) {
         major = minor = revision = 0;
+      } else if (Target.isOSBinFormatWasm()) {
```

Comment:
This seems wrong and is creating a branch which is going to cause more incorrect checks in the future.  This should be part of the check in the previous condition: the os is `none` should set `major` and `minor` to 0.